lib/administrate/field/has_many.rb

```diff
@@ -25,7 +25,9 @@ def associated_resource_options
       end
 
       def selected_options
-        data && data.map { |object| object.send(primary_key) }
+        return if data.empty?
```

Comment:
In order to keep the same behavior as before, I added this guard clause. 

Now that `data` won't be `nil`, we can let the `map` method run and it will return an empty array.

So in the case of an empty collection, do we want this method to return `nil` or `[]`?